### PR TITLE
Updated nudge email for new reference usage

### PR DESCRIPTION
**Pull request by MylesJarvis**
## Context

We have a nudge that is sent when a candidate is stuck on references (has completed all other sections) and has been inactive for 7 days. At the moment, this copy references the old world of references. This draft attempts to bring it back to the new way of reference collection, drawing on copy used in the [referee flow PR](https://github.com/DFE-Digital/apply-for-teacher-training/pull/7489/files).

## Changes proposed in this pull request

Update copy to reflect the new reference process.

## Guidance to review

- Does the copy make sense
- Do we ned to remove the term 'referee'? I think we have done elsewhere.

## Link to Trello card

https://trello.com/c/XdGHvt5E/712-amend-copy-and-re-enable-our-unsubmitted-application-with-just-references-left-to-complete-nudge

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by carlosmartinez** [comment]
@MylesJarvis is this waiting on feedback?

**Comment by MylesJarvis** [comment]
> @MylesJarvis is this waiting on feedback?

@carlosmartinez yep I've asked @johndoates to take a look today.

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:5]:
```suggestion
You’ll need to give details of 2 people who can give you a reference. They’ll only be contacted if you accept an offer on a course.
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:3]: Do we think it's ok that it'll say this even in the (unlikely) scenario that you’ve added 2+ reference but just not marked the section as complete?

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (frankieroberto) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:17]: Could drop all this, or just simplify it to "They’ll be asked if they know any reason why you should not work with children." to match the web interface?

Comment (MylesJarvis) [reply app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:17]: I'd be happy mirroring the web interface - will defer to @johndoates. I lifted this from the referee request emails as that was 'approved' content.

**Comment by MylesJarvis** [comment]
Turns out that this is gonna need some dev input due to the way we select for it. I suggest we turn this nudge off in the feature flag dashboard and come back to it.

Relevant code here (from app/mailers/candidate_mailer.rb):

`def nudge_unsubmitted_with_incomplete_references_template_name(application_form)
    number_of_references_received = application_form.application_references.select(&:feedback_provided?).count
    number_of_references_requested = application_form.application_references.select(&:feedback_requested?).count

    if (number_of_references_requested + number_of_references_received).zero?
      :no_references
    elsif number_of_references_received.zero? &&
          number_of_references_requested < ApplicationForm::REQUIRED_REFERENCE_SELECTIONS
      :one_requested_reference
    elsif number_of_references_received < ApplicationForm::REQUIRED_REFERENCE_SELECTIONS &&
          number_of_references_requested.zero?
      :one_received_reference
    elsif number_of_references_received < ApplicationForm::REQUIRED_REFERENCE_SELECTIONS &&
          number_of_references_requested < ApplicationForm::REQUIRED_REFERENCE_SELECTIONS
      :two_references
    end
  end`

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (MylesJarvis) [reply app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:3]: 'You have not yet completed the reference section of your teacher training application'?

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:16]:
```suggestion
  - say if they know any reason why you should not work with children
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:17]:
```suggestion
  - give factual information which can be used to check your application
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:3]:
```suggestion
You have not yet completed the references section of your teacher training application.
```

Comment (johndoates) [reply app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:3]: I've made a suggestion. It's a bit more technical sounding ("the references section") but more accurate.

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [reply app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:17]: I've made a couple of suggestions to make it better match the latest wording in other places. 

I wondered whether we need the 'who to choose' section at all. There could be some benefit if the candidate is hesitant to click through because they don't have someone in mind, but we aren't giving them all the details. For example, they may have a couple of professionals in mind but when they click through they discover they need an academic referee.

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:9]:
```suggestion
# Who to choose to give you a reference
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:11]:
```suggestion
You could choose someone like a manager, a university tutor or a client.
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:13]:
```suggestion
In their reference, they will be asked to:
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:5]:
```suggestion
You need to give details of 2 people who can give you a reference. They’ll only be contacted if you accept an offer on a course.
```

File: app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb
Comment (johndoates) [inline app/views/candidate_mailer/nudge_unsubmitted_with_incomplete_references/no_references.text.erb:7]:
```suggestion
[Sign into your account to complete the references section.](<%= candidate_magic_link(@application_form.candidate) %>).
```

**Comment by MylesJarvis** [comment]
Okay I've just committed @johndoates' suggestions. Next step is a convo with @carlosmartinez on the feature flag and disabling it for the new cycle.

**Comment by carlosmartinez** [comment]
@MylesJarvis ready to roll?

**Comment by MylesJarvis** [comment]
@carlosmartinez yes please. Shall we chat about how we approach these tests?

**Review by carlosmartinez — APPROVED** [review]

**Pull request merged**
